onnx/defs/generator/defs.cc

```diff
@@ -33,69 +33,36 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, tensor_proto);
         }));
 
-static const char* ConstantLike_ver9_doc = R"DOC(
-Generate a tensor with specific constant value. The value can be specified by the 'value' 
-attribute. The shape of the output tensor is the same as the input tensor, if the input 
-tensor is provided, or the shape provided in the 'shape' attribute (if both are provided, 
-the input tensor shape takes precendence). The data type can be specified by the 'dtype' 
-argument. If 'dtype' is not specified, then the type of input tensor is used. If input 
-tensor is also not specified, then the type defaults to 'float'.
-
-The 'dtype' argument must be one of the data types specified in the 'DataType' enum field in the
-TensorProto message and be valid as an output type.
+static const char* ConstantAsShape_ver9_doc = R"DOC(
+Generate a tensor with given value and shape.
 )DOC";
 
 ONNX_OPERATOR_SET_SCHEMA(
-    ConstantLike,
+    ConstantAsShape,
     9,
     OpSchema()
-        .SetDoc(ConstantLike_ver9_doc)
+        .SetDoc(ConstantAsShape_ver9_doc)
         .Attr(
             "value",
-            "(Optional) The value for the elements of the output tensor.",
-            AttributeProto::FLOAT,
-            0.0f)
-        .Attr(
-            "dtype",
-            "(Optional) The data type for the elements of the output tensor. If not specified,"
-            "the data type of the input tensor T1 is used. If input tensor T1 is also not "
-            "specified, then output tensor type defaults to 'float'.",
-            AttributeProto::INT,
-            OPTIONAL)
-        .Attr(
-            "shape",
-            "(Optional) The shape of the output tensor. If input tensor T1 is provided, then"
-            " 'shape' attribute is ignored and the output follows the shape of the input."
-            " One of either input tensor T1 or 'shape' attribute must be provided.",
-            AttributeProto::INTS,
-            OPTIONAL)
+            "(Optional) The value of the output elements."
+            "Should be a one-element tensor, "
+            "If 'dtype' is not given, the datatype of "
```

Comment:
Did you mean: "The value for this attribute defaults to 0."?

Reply:
@zrphercule - Minor point - this probably needs some clarity. I think we want the datatype of the output tensor to be the same as the type of one-element `value` tensor, correct? If yes, then this can be worded better. For one, there's no `dtype` attribute.

Reply:
Oh yes, but the value tensor is optional. Let's say if value is not given, then by default the output dtype is float32.